File: java/solutions/02-intermediate/ex068_junit_parameterized/PrimeChecker.java
package fewolearning.exercises.intermediate.ex068_junit_parameterized;

/*
Exercise 068 - JUnit parameterized (reference solution).
*/
public final class PrimeChecker {
    private PrimeChecker() {
    }

    public static boolean isPrime(int candidate) {
        if (candidate < 2) {
            return false;
        }
        for (int divisor = 2; (long) divisor * divisor <= candidate; divisor++) {
            if (candidate % divisor == 0) {
                return false;
            }
        }
        return true;
    }
}

File: kotlin/solutions/02-intermediate/ex060_junit_parameterized/PrimeChecker.kt
package fewolearning.exercises.intermediate.ex060_junit_parameterized

/** Checks whether [candidate] is a prime number by trial division up to its square root. */
fun isPrime(candidate: Int): Boolean {
    if (candidate < 2) return false
    if (candidate == 2) return true
    if (candidate % 2 == 0) return false

    var divisor = 3
    while (divisor * divisor <= candidate) {
        if (candidate % divisor == 0) return false
        divisor += 2
    }
    return true
}
